### ISSUE TITLE
uninitialized constant Chef::Resource::UlimitRule when using ulimit_domain

### ISSUE THREAD

**Issue by ghost** [bmhatfield/chef-ulimit#62]
### Cookbook version
v0.4.0 from Supermarket

### Chef-client version
12.17.44 (inside Docker via kitchen-dokken)

### Platform Details
CentOS 7.4.1708
Running under kitchen-dokken via ChefDK
```
$ chef --version
Chef Development Kit Version: 2.0.28
chef-client version: 13.2.20
delivery version: master (17c1b0fed9be4c70f69091a6d21a4cbf0df60a23)
berks version: 6.2.0
kitchen version: 1.16.0
inspec version: 1.31.1
```

### Scenario:
Attempting to set limits for HDFS (nofile, nproc) via this cookbook.

### Steps to Reproduce:
`.kitchen.dokken.yml`:
```yaml
driver:
  name: dokken
  privileged: true # systemd, docker, and sysctl, oh my!
  # Set this to configure the Chef version to test
  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>

transport:
  name: dokken

provisioner:
  name: dokken
  client_rb:
    treat_deprecation_warnings_as_errors: true
    resource_cloning: false
  # chef_log_level: debug

platforms:
  - name: centos-7
    driver:
      image: centos:7
      platform: rhel
      pid_one_command: /usr/lib/systemd/systemd
      intermediate_instructions:
        - RUN yum -y install lsof which systemd-sysv initscripts wget net-tools libselinux-utils make
suites:
  - name: default
    run_list:
    - recipe[cookbook::limits]
```
`cookbook::limits.rb`
```ruby
include_recipe 'ulimit::default'

# loosen limits on files and processes
ulimit_domain 'hdfs' do
  rule do
    item 'nofile'
    type '-'
    value 'unlimited'
  end
  rule do
    item 'nproc'
    type '-'
    value 32768
  end
end
```

Steps to reproduce:
```bash
KITCHEN_LOCAL_YAML=.kitchen.dokken.yml kitchen converge default
```

### Expected Result:
Convergence! A file `hdfs.conf` with my limits set.

### Actual Result:
```
  * ulimit_domain[hdfs] action create

    ================================================================================
    Error executing action `create` on resource 'ulimit_domain[hdfs]'
    ================================================================================

    NameError
    ---------
    uninitialized constant Chef::Resource::UlimitRule

    Cookbook Trace:
    ---------------
    /opt/kitchen/cache/cookbooks/ulimit/providers/domain.rb:6:in `block in load_current_resource'
    /opt/kitchen/cache/cookbooks/ulimit/providers/domain.rb:5:in `map!'
    /opt/kitchen/cache/cookbooks/ulimit/providers/domain.rb:5:in `load_current_resource'

    Resource Declaration:
    ---------------------
    # In /opt/kitchen/cache/cookbooks/cookbook/recipes/limits.rb

     20: ulimit_domain 'hdfs' do
     21:   rule do
     22:     item 'nofile'
     23:     type '-'
     24:     value 'unlimited'
     25:   end
     26:   rule do
     27:     item 'nproc'
     28:     type '-'
     29:     value 32768
     30:   end
     31: end

    Compiled Resource:
    ------------------
    # Declared in /opt/kitchen/cache/cookbooks/cookbook/recipes/limits.rb:20:in `from_file'

    ulimit_domain("hdfs") do
      subresource_rules [[nil, #<Proc:0x0000000349f680@/opt/kitchen/cache/cookbooks/cookbook/recipes/limits.rb:21>], [nil, #<Proc:0x0000000349f900@/opt/kitchen/cache/cookbooks/cookbook/recipes/limits.rb:26>]]
      action [:create]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :ulimit_domain
      cookbook_name "cookbook"
      recipe_name "limits"
      filename "hdfs.conf"
    end

    Platform:
    ---------
    x86_64-linux


Running handlers:
[2017-11-09T19:49:28+00:00] ERROR: Running exception handlers
Running handlers complete
[2017-11-09T19:49:28+00:00] ERROR: Exception handlers complete
Chef Client failed. 1 resources updated in 24 seconds
[2017-11-09T19:49:28+00:00] FATAL: Stacktrace dumped to /opt/kitchen/cache/chef-stacktrace.out
[2017-11-09T19:49:28+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-11-09T19:49:28+00:00] ERROR: ulimit_domain[hdfs] (cookbook::limits line 20) had an error: NameError: uninitialized constant Chef::Resource::UlimitRule
[2017-11-09T19:49:28+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Converge failed on instance <default-centos-7>.  Please see .kitchen/logs/default-centos-7.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by wolf31o2** [comment]
Removing (merging) above GitHub account, so commenting to get this into my notifications.

**Comment by jshrack-ssi** [comment]
I am seeing a similar issue on CentOS 6.5 with Chef 13.6.4.

**Comment by jshrack-ssi** [comment]
@bmhatfield Any chance of fixing this issue? This is a major blocker for users running Chef 13. So far all my testing shows this is specifically related to Chef 13. Looks like a resources re-write is needed.

**Comment by tas50** [comment]
This should be fixed in the 1.0 release that's up on the Supermarket now. We have kitchen tests now and everything is green.
